src/components/app.jsx

```diff
@@ -35,12 +42,23 @@ const RoutedEditor = withRouter(props => {
   )
 })
 
-const PrivateContext = () => (
-  <Switch>
-    <Route path="/n/:id" component={RoutedEditor} />
-    <Route path="/" component={List} />
-  </Switch>
-)
+const PrivateContext = () => {
+  const location = useLocation()
+  const background = location.state?.background
+
+  return (
+    <>
+      <Switch>
+        <Route path="/n/:id" component={RoutedEditor} />
+        <Route path="/" component={List} />
+      </Switch>
+
+      {background && (
```

Comment:
This is how we handle the modal as route pattern. Cf https://stackblitz.com/github/remix-run/react-router/tree/main/examples/modal?file=src/App.tsx